### PR TITLE
feat: implement Clone Hero .sng export

### PR DESCRIPTION
**Pull request by AshMartian**
This PR adds a new 'Export' feature to the toolbar to bundle and pack songs into Clone Hero `.sng` packages.

### Features
1. **Visual UI Controls**: Added an 'Export' button in the main toolbar that displays a dark-themed modal window to select output directories, customize output filenames, and decide whether to save edits before exporting.
2. **Settings Integration**: Remembers and persists the last output directory used (with a 'Reset to Default' shortcut button to fallback to the auto-chart output folder).
3. **Sequential Stream Packaging**: Replaces memory-heavy `Buffer.concat` with a low-RAM sequential `WriteStream`, lowering the memory footprint of multi-track stem packages.
4. **Atomic Swap File Writes**: Generates and writes package data to a temporary file (`*.sng.tmp`) first and swaps atomically to avoid package corruptions.
5. **Format Validation**: Automatically ignoring editor metadata files (`video.json`, `audio.json`, and `venue.json`) while verifying that playable chart files are present (and filenames don't exceed length limits).
6. **Reveal File in Explorer**: Success callouts include a 'Show in Folder' reveal action.
7. **Comprehensive Tests**: Includes detailed Vitest unit tests covering edge-case validations and packing offsets.

### References & Acknowledgments
The binary layout and encryption logic are built using the official [Onyx Reference Implementation](https://github.com/mtolly/onyx):
* [SNG.hs](https://github.com/mtolly/onyx/blob/master/haskell/packages/onyx-lib/src/Onyx/CloneHero/SNG.hs): Outlines the `.sng` index serialization layout, string boolean normalizations, and XOR encryption pattern.
* [Handle.hs](https://github.com/mtolly/onyx/blob/master/haskell/packages/onyx-handle/src/Onyx/Util/Handle.hs): Demonstrates that sub-handles use file-relative offsets (`shTell`), clarifying that the XOR keystream is 0-indexed relative to each file payload.
* [CloneHero.hs](https://github.com/mtolly/onyx/blob/master/haskell/packages/onyx-lib/src/Onyx/Build/CloneHero.hs): Clarified typical asset gather/ignore rules (e.g. ignoring `song.ini` in the data stream since metadata covers it).

image
image

### CODE REVIEW

File: src/main/sngPacker.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+      }
+    }
+
+    // Wait for the stream to finish writing
+    await new Promise<void>((resolve, reject) => {
+      writeStream.end(() => {
+        resolve()
+      })
+      writeStream.on('error', reject)
+    })
```
Comment (opria123) [inline src/main/sngPacker.ts:271]: The `'error'` listener is only attached inside the final `writeStream.end(...)` promise (line 234), so an error during the payload loop above has no handler — risking an unhandled `'error'` event (crash) and an orphaned `.sng.tmp`. The drain wait on lines 223–225 can also hang forever if the stream errors while awaiting `'drain'`.

Attach the error handler once up front (right after `createWriteStream` on line 200) and race it against the drain wait:

```ts
const writeStream = createWriteStream(tempPath)
const streamError = new Promise<never>((_, reject) =>
  writeStream.once('error', reject)
)
// ...
const canWrite = writeStream.write(encrypted)
if (!canWrite) {
  await Promise.race([
    new Promise<void>((resolve) => writeStream.once('drain', resolve)),
    streamError
  ])
}
```

Any write failure then rejects promptly and still hits the `catch` that unlinks the temp file.

File: src/main/sngPacker.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    // Write headers
+    writeStream.write(headerBuffer)
+    writeStream.write(metaBuffer)
+    writeStream.write(indexBuffer)
+    writeStream.write(payloadHeaderBuffer)
+
+    // Write file payloads sequentially (streaming style, low RAM profile)
+    for (const entry of fileEntries) {
+      const filePath = join(songDir, entry.name)
+      const rawData = await readFile(filePath)
```
Comment (opria123) [inline src/main/sngPacker.ts:219]: The index records each entry's `size` (and therefore all offsets and the data-section length) from `statSync()` during the gather phase, but the payload is read here later via `readFile`. If a file's size changes in between, the written bytes no longer match the recorded offsets/length and the `.sng` is silently malformed — no error, just a bad package. This is especially likely because the export flow saves edits right before packing (the default option), so files are written moments before they're read.

Guard against it:

```ts
const rawData = await readFile(filePath)
if (rawData.length !== entry.size) {
  throw new Error(`File changed size during export: ${entry.name}`)
}
```

The temp-file cleanup in the `catch` then prevents a partial `.sng` from being left behind.

File: src/renderer/src/components/ExportModal.css [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+  border: 1px solid var(--border-color);
+  background: rgba(255, 255, 255, 0.01);
+  transition: all 0.2s ease;
+  cursor: pointer;
+  position: relative;
+}
+
+.export-format-card.active {
+  border-color: #3b82f6; /* Accent color */
+  background: rgba(59, 130, 246, 0.05);
```
Comment (opria123) [inline src/renderer/src/components/ExportModal.css]: These hardcoded hex values diverge from the rest of the app, which uses theme tokens like `var(--accent-color)` / `var(--text-secondary)` (the `.export-modal-primary` button in this same file already does). The result is a blue active-card glow next to the app's amber accent button, and these won't track future theme changes.

Same issue appears at:
- line 156 (`background: #3b82f6`)
- line 198 (`color: #ef4444`)
- line 270 (`color: #10b981`)
- lines 304 / 340 (`#3b82f6`)
- line 435 (`color: #f59e0b`)

Please swap them for the existing CSS variables so the export UI stays consistent with the rest of the editor.

**Review by opria123 — COMMENTED** [review]
Nice feature and solid test coverage. Two correctness items I'd like fixed before merge (write-stream error handling and an offset/size mismatch that can silently corrupt the package), plus one styling-consistency note. Details inline.

**Review by opria123 — CHANGES_REQUESTED** [review]

**Comment by AshMartian** [comment]
All requested review items have been resolved:
1. **WriteStream Error Handling**: The error listener is now attached up front and raced during drain/finish waits, preventing any orphaned `.sng.tmp` files or crashes.
2. **File Size Change Check**: Added size checks right before payload reads to ensure files have not changed size since the index was computed.
3. **Theme Variables & Consistency**: Refactored all hardcoded hex and rgba color coordinates in the CSS to follow the editor's official CSS theme tokens (like `var(--accent-color)`, `var(--success-color)`, `var(--danger-color)`, and `var(--warning-color)`).
